packages/node-core/src/indexer/blockDispatcher/base-block-dispatcher.ts

```diff
@@ -208,6 +208,12 @@ export abstract class BaseBlockDispatcher<Q extends IQueue, DS> implements IBloc
       // Flush all data from cache and wait
       await this.storeCacheService.flushCache(false, true);
     }
+
+    if (!(await this.projectService.hasDataSourcesAfterHeight(height))) {
```

Comment:
I'm not sure this is right. What happens if we run into bypassblocks at the next height?

Reply:
maybe we can do this check with datasources map to make sure we are not exiting prematurely